src/StructuredLogViewer.Core/Timeline/Lane.cs

```diff
@@ -8,18 +9,29 @@ public class Lane
         public List<Block> Blocks { get; set; } = new List<Block>();
 
         public void Add(Block block)
+        {
+            if (IsValid(block))
+                Blocks.Add(block);
+        }
+
+        public void AddRange(IEnumerable<Block> blocks)
+        {
+            Blocks.AddRange(blocks.Where(b => IsValid(b)));

```

Comment:
slightly worried about perf here. Is this a hotspot? Perhaps worth rewriting as a simple foreach loop and call Blocks.Add?

Reply:
Removed as it is unused.